errtracker/errtracker.go

```diff
@@ -133,7 +209,25 @@ func Report(snap, errMsg, dupSig string, extra map[string]string) (string, error
 	extra["ProblemType"] = "Snap"
 	extra["Snap"] = snap
 
-	return report(errMsg, dupSig, extra)
+	// check if we haven't already reported this error
+	db, err := newReportsDB(dirs.ErrtrackerDbDir)
+	if err != nil {
+		return "", err
```

Comment:
Shouldn't we just swallow the error here? If there's some storage issue, we'll get an error back from `newReportsDB`, but it would still make sense to try and report the original error upstream.

Reply:
Good point.